src/lib.rs

```diff
@@ -2039,6 +1994,65 @@ impl_float_const!(OrderedFloat, OrderedFloat);
 // Float constants are not NaN.
 impl_float_const!(NotNan, |x| unsafe { NotNan::new_unchecked(x) });
 
+// canonical raw bit patterns (for hashing)
+
+mod hash_internals {
+    pub trait SealedTrait: Copy + num_traits::float::FloatCore {
+        type Bits: core::hash::Hash;
+
+        const CANONICAL_NAN_BITS: Self::Bits;
+
+        fn canonical_bits(self) -> Self::Bits;
+    }
+
+    impl SealedTrait for f32 {
+        type Bits = u32;
+
+        const CANONICAL_NAN_BITS: u32 = 0x7fc00000;
+
+        fn canonical_bits(self) -> u32 {
+            // -0.0 + 0.0 == +0.0 under IEEE754 roundTiesToEven rounding mode,
+            // which Rust guarantees. Thus by adding a positive zero we
+            // canonicalize signed zero without any branches in one instruction.
+            (self + 0.0).to_bits()
+        }
+    }
+
+    impl SealedTrait for f64 {
+        type Bits = u64;
+
+        const CANONICAL_NAN_BITS: u64 = 0x7ff8000000000000;
+
+        fn canonical_bits(self) -> u64 {
+            (self + 0.0).to_bits()
+        }
+    }
+}
+
+/// The built-in floating point types `f32` and `f64`.
+///
+/// This is a "sealed" trait that cannot be implemented for any other types.
+pub trait PrimitiveFloat: hash_internals::SealedTrait {}
+impl PrimitiveFloat for f32 {}
+impl PrimitiveFloat for f64 {}
+
+impl<T: PrimitiveFloat> Hash for OrderedFloat<T> {
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        let bits = if self.0.is_nan() {
+            T::CANONICAL_NAN_BITS
+        } else {
+            self.0.canonical_bits()
+        };
+        bits.hash(hasher);
+    }
+}
```

Comment:
This breaks the original hash result because it doesn't call `raw_double_bits` anymore.

Our use case depends on stable hash results. @mbrubeck would you give some background why change the result? (I can understand we implement only for f32/f64)

Reply:
Sorry about that. The main reason for that change was performance. (See #142 for details.)